src/Symfony/Component/DependencyInjection/Compiler/ServiceLocatorTagPass.php

```diff
@@ -91,7 +97,11 @@ public static function register(ContainerBuilder $container, array $refMap, $cal
             ->setPublic(false)
             ->addTag('container.service_locator');
 
-        if (!$container->has($id = 'service_locator.'.ContainerBuilder::hash($locator))) {
+        if (null !== $callerId && $container->hasDefinition($callerId)) {
+            $locator->setBindings($container->getDefinition($callerId)->getBindings());
+        }
+
+        if (!$container->hasDefinition($id = 'service_locator.'.ContainerBuilder::hash($locator))) {
```

Comment:
are bindings taken into account by the hashing ? If no, that would cause issues.

Reply:
correct, hash() uses serialize() so all the state is hashed